webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -28,10 +28,38 @@ import { MessageDuration } from "../message-duration";
 import * as monaco from 'monaco-editor'
 
 let stateCache = {};
+const WHITESPACE_256 = String.fromCharCode.apply(null, new Array(256).fill(0x20));
 let lastFile;
 //Unsupported DS types
 const unsupportedTypes: Array<string> = ['G', 'B', 'C', 'D', 'I', 'R'];
 
+function isContentValidForDataset(content: string[], datasetAttrs: DatasetAttributes): boolean | string {
+  //TODO: validation of record length that is aware of how DBCS will effect actual length
+  //FB must have exactly lrecl, VB must have no more than lrecl. content cant be undefined.
+  const MAX_CONTENT_LENGTH = 5242880;
+  if (!content) {
+    return 'No Content';
+  }
+  let isFixedRecordLength = datasetAttrs.recfm.recordLength === 'F';
+  let maxRecordLen = datasetAttrs.dsorg.maxRecordLen;
+  for (let i = 0; i < content.length; i++) {
+    let record = content[i];
+    if (record.length > maxRecordLen) {
+      return `Line ${i} exceeds record length limit of ${maxRecordLen}`;
+    } else if (isFixedRecordLength && record.length < maxRecordLen) {
+      let whitespace = WHITESPACE_256;
+      while (whitespace.length < maxRecordLen) {
+        whitespace += WHITESPACE_256;
+      }
+      content[i] = record+whitespace.substring(record.length,maxRecordLen);
```

Comment:
I know this came from my code but you should check if this is even needed anymore. server-side padding may have been implemented now which would be superior.

Reply:
If I am not wrong we might still require this because if the user is modifying the dataset such that the record length is less than the LRECL then the space padding is not added by the server.